File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.py
from collections import deque

n, m = map(int, input().split())
targets = list(map(int, input().split()))

dq = deque([i + 1 for i in range(n)])

ans = 0
for target in targets:
    rotate_cnt = 0
    while(dq[0] != target):
        rotate_cnt += 1
        dq.rotate(1)
    ans = ans + min(rotate_cnt, len(dq) - rotate_cnt)
    dq.popleft()
    
print(ans)

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.cc
#include <iostream>
#include <deque>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n, m;
    cin >> n >> m;
    
    deque<int> dq;
    for (int i = 0; i < n; i++) {
        dq.push_back(i + 1);
    }
    
    int ans = 0;
    for (int i = 0; i < m; i++) {
        int target;
        cin >> target;
        
        int rotate_cnt = 0;
        while(dq.front() != target) {
            dq.push_back(dq.front());
            dq.pop_front();
            rotate_cnt++;
        }
        ans += min(rotate_cnt, (int) dq.size() - rotate_cnt);
        dq.pop_front();
    }
    cout << ans;
    return (0);
}